src/client/eppo-client.ts

```diff
@@ -182,21 +209,42 @@ export default class EppoClient {
    * @param defaultValue default value to return if the subject is not part of the experiment sample
    * @returns a boolean variation value if the subject is part of the experiment sample, otherwise the default value
    */
-  public getBooleanAssignment(
+  public getBooleanAssignment = (
     flagKey: string,
     subjectKey: string,
     subjectAttributes: Record<string, AttributeType>,
     defaultValue: boolean,
-  ): boolean {
-    return (
-      this.getAssignmentVariation(
-        flagKey,
-        subjectKey,
-        subjectAttributes,
-        EppoValue.Bool(defaultValue),
-        VariationType.BOOLEAN,
-      ).boolValue ?? defaultValue
+  ) => this.getBooleanAssignmentDetails(flagKey, subjectKey, subjectAttributes, defaultValue).value;
+
+  /**
+   * Maps a subject to a boolean variation for a given experiment and provides additional details about the
+   * variation assigned and the reason for the assignment.
+   *
+   * @param flagKey feature flag identifier
+   * @param subjectKey an identifier of the experiment subject, for example a user ID.
+   * @param subjectAttributes optional attributes associated with the subject, for example name and email.
+   * @param defaultValue default value to return if the subject is not part of the experiment sample
+   * The subject attributes are used for evaluating any targeting rules tied to the experiment.
+   * @returns an object that includes the variation value along with additional metadata about the assignment
+   * @public
+   */
+  public getBooleanAssignmentDetails(
+    flagKey: string,
+    subjectKey: string,
+    subjectAttributes: Record<string, AttributeType>,
+    defaultValue: boolean,
+  ): AssignmentDetails<boolean> {
+    const { eppoValue, flagEvaluationDetails } = this.getAssignmentVariation(
+      flagKey,
+      subjectKey,
+      subjectAttributes,
+      EppoValue.Bool(defaultValue),
+      VariationType.BOOLEAN,
     );
+    return {
+      value: eppoValue.boolValue ?? defaultValue,
```

Comment:
sorta unrelated, but I think we should have `getAssignmentVariation` return value already taking into account the default value, to avoid this `value: eppoValue.boolValue ?? defaultValue,` repetitiveness

Reply:
👍 I can do that

Reply:
Unfortunately, we're going to get some weirdness by moving the `?? defaultValue` to a centralized place due to how we're passing explicit types (e.g. `EppoValue.Bool`, `VariationType.Boolean`)  rather than using TypeScript generics (e.g. `getAssignmentVariation<boolean>(...`). We're passing the types for vanilla JS support. Given all that, I think the `?? defaultValue` bit is probably the most elegant way to handle this. 

<img width="1441" alt="Screenshot 2024-06-24 at 5 18 22 AM" src="https://github.com/Eppo-exp/js-client-sdk-common/assets/1248641/823b6be7-1c33-4698-8efe-8f7b9fadae29">